### PR TITLE
GEOMESA-826 Add kafka datastore quickstart

### PR DESCRIPTION
**Pull request by Afterglow375**
* Added KafkaDataStore quickstart tutorial to geomesa-examples module

Signed-off-by: Alex Tatusko <atatusko@ccri.com>

### CODE REVIEW

File: geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public class AccumuloQuickStart {
+    static String INSTANCE_ID = "instanceId";
```
Comment (mmatz-ccri) [inline geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java]: make final

File: geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+import org.geotools.feature.FeatureIterator;
+import org.geotools.feature.SchemaException;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.filter.text.cql2.CQL;
+import org.geotools.filter.text.cql2.CQLException;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.locationtech.geomesa.accumulo.data.AccumuloFeatureStore;
+import org.locationtech.geomesa.accumulo.index.Constants;
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes$;
```
Comment (mmatz-ccri) [inline geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java]: use org.locationtech.geomesa.utils.interop.SimpleFeatureTypes instead

File: geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                "What:java.lang.Long",     // some types require full qualification (see DataUtilities docs)
+                "When:Date",               // a date-time field is optional, but can be indexed
+                "*Where:Point:srid=4326",  // the "*" denotes the default geometry (used for indexing)
+                "Why:String"               // you may have as many other attributes as you like...
+        );
+
+        // create the bare simple-feature type
+        String simpleFeatureTypeSchema = Joiner.on(",").join(attributes);
+        SimpleFeatureType simpleFeatureType =
+                SimpleFeatureTypes$.MODULE$.createType(simpleFeatureTypeName, simpleFeatureTypeSchema);
```
Comment (mmatz-ccri) [inline geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java]: simplifies with org.locationtech.geomesa.utils.interop.SimpleFeatureTypes

File: geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+import org.geotools.feature.SchemaException;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.filter.text.cql2.CQL;
+import org.geotools.filter.text.cql2.CQLException;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.locationtech.geomesa.accumulo.data.AccumuloFeatureStore;
+import org.locationtech.geomesa.accumulo.index.Constants;
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes$;
+import org.locationtech.geomesa.utils.text.WKTUtils$;
```
Comment (mmatz-ccri) [inline geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java]: create org.locationtech.geomesa.utils.interop.WKTUtils?

File: geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+        return featureCollection;
+    }
+
+    static void insertFeatures(String simpleFeatureTypeName,
+                               DataStore dataStore,
+                               FeatureCollection featureCollection)
+            throws IOException {
+
+        FeatureStore featureStore = (AccumuloFeatureStore)dataStore.getFeatureSource(simpleFeatureTypeName);
```
Comment (mmatz-ccri) [inline geomesa-examples/accumulo-quickstart/src/main/java/org/locationtech/geomesa/examples/AccumuloQuickStart.java]: cast to SimpleFeatureStore instead

File: geomesa-examples/kafka-quickstart/src/main/java/org/locationtech/geomesa/examples/KafkaQuickStart.java [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public class KafkaQuickStart {
+    public static String KAFKA_BROKER_PARAM = "brokers";
```
Comment (mmatz-ccri) [inline geomesa-examples/kafka-quickstart/src/main/java/org/locationtech/geomesa/examples/KafkaQuickStart.java]: make final

File: geomesa-examples/kafka-quickstart/src/main/java/org/locationtech/geomesa/examples/KafkaQuickStart.java [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+            producerDS.createSchema(preppedOutputSft);
+
+        System.out.println("Register KafkaDataStore in GeoServer (Press enter to continue)");
+        System.in.read();
+
+        // the live consumer must be created before the producer writes features
+        // in order to read streaming data.
+        // i.e. the live consumer will only read data written after its instantiation
+        SimpleFeatureSource consumerFS = consumerDS.getFeatureSource(sftName);
+        KafkaProducerFeatureStore producerFS = (KafkaProducerFeatureStore) producerDS.getFeatureSource(sftName);
```
Comment (mmatz-ccri) [inline geomesa-examples/kafka-quickstart/src/main/java/org/locationtech/geomesa/examples/KafkaQuickStart.java]: use SimpleFeatureStore instead

File: geomesa-examples/kafka-quickstart/src/main/java/org/locationtech/geomesa/examples/KafkaQuickStart.java [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+        producerDS.createSchema(replaySFT);
+        SimpleFeatureSource replayConsumerFS = consumerDS.getFeatureSource(replaySFT.getName());
+
+        // querying for the state of SimpleFeatures approximately 5 seconds before the replayEnd.
+        // the ReplayKafkaConsumerFeatureSource will build the state of SimpleFeatures
+        // by processing all of the messages that were sent in between queryTime-readBehind and queryTime.
+        // only the messages in between replayStart and replayEnd are cached.
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        Date queryTime = replayEnd.minus(5000).toDate();
+        featureCollection = replayConsumerFS.getFeatures(ff.equals(ff.property("KafkaLogTime"), ff.literal(queryTime)));
```
Comment (mmatz-ccri) [inline geomesa-examples/kafka-quickstart/src/main/java/org/locationtech/geomesa/examples/KafkaQuickStart.java]: could use ReplayTimeHelper.toFilter(queryTime) to create the filter

File: geomesa-examples/accumulo-quickstart/pom.xml [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+            <artifactId>gt-data</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.geotools</groupId>
+            <artifactId>gt-epsg-hsql</artifactId>
+            <version>${gt.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.objenesis</groupId>
+            <artifactId>objenesis</artifactId>
```
Comment (AndrewAnnex) [inline geomesa-examples/accumulo-quickstart/pom.xml]: [picky] add version numbers to each of the dependencies via params set in the main pom (like you have for geotools above)

**Comment by mmatz-ccri** [comment]
modules "*-quickstart" are missing"geomesa-" prefix used by all other modules, including other sub-modules

**Comment by anthonyccri** [comment]
Merged.

**Pull request closed without merging**